pkg/virt-config/configuration_test.go

```diff
@@ -0,0 +1,630 @@
+		clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
+			MigrationConfiguration: &v1.MigrationConfiguration{
+				ParallelOutboundMigrationsPerNode: &parallelOutboundMigrationsPerNode,
+				ParallelMigrationsPerCluster:      &parallelMigrationsPerCluster,
+				BandwidthPerMigration:             &bandwidthPerMigration,
+				ProgressTimeout:                   &progressTimeout,
+				CompletionTimeoutPerGiB:           &completionTimeoutPerGiB,
+				UnsafeMigrationOverride:           &trueValue,
+				AllowAutoConverge:                 &trueValue,
+			},
+		})
+
+		result := clusterConfig.GetMigrationConfiguration()
+		Expect(*result.ParallelOutboundMigrationsPerNode).To(BeNumerically("==", 10))
+		Expect(*result.ParallelMigrationsPerCluster).To(BeNumerically("==", 20))
+		bandwidth := *result.BandwidthPerMigration
+		Expect(bandwidth.String()).To(Equal("110Mi"))
+		Expect(*result.ProgressTimeout).To(BeNumerically("==", 5))
+		Expect(*result.CompletionTimeoutPerGiB).To(BeNumerically("==", 5))
+		Expect(*result.UnsafeMigrationOverride).To(BeTrue())
+		Expect(*result.AllowAutoConverge).To(BeTrue())
+	})
+
+	It("Should return defaults if parts of the config are not set", func() {
+		parallelOutboundMigrationsPerNode := uint32(10)
+		clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
+			MigrationConfiguration: &v1.MigrationConfiguration{
+				ParallelOutboundMigrationsPerNode: &parallelOutboundMigrationsPerNode,
+			},
+		})
+
+		result := clusterConfig.GetMigrationConfiguration()
+		Expect(*result.ParallelOutboundMigrationsPerNode).To(BeNumerically("==", 10))
+		Expect(*result.ParallelMigrationsPerCluster).To(BeNumerically("==", 5))
+		Expect(result.BandwidthPerMigration.String()).To(Equal("0"))
+	})
+
+	It("Should update the config if a newer version is available", func() {
+		oldValue := uint32(10)
+		clusterConfig, _, kvInformer := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
+			MigrationConfiguration: &v1.MigrationConfiguration{
+				ParallelOutboundMigrationsPerNode: &oldValue,
+			},
+		})
+		result := clusterConfig.GetMigrationConfiguration()
+		Expect(*result.ParallelOutboundMigrationsPerNode).To(BeEquivalentTo(10))
+
+		newValue := uint32(9)
+		kv := &v1.KubeVirt{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kubevirt",
+				Namespace: "kubevirt",
+			},
+			Spec: v1.KubeVirtSpec{
+				Configuration: v1.KubeVirtConfiguration{
+					MigrationConfiguration: &v1.MigrationConfiguration{
+						ParallelOutboundMigrationsPerNode: &newValue,
+					},
+				},
+			},
+			Status: v1.KubeVirtStatus{
+				Phase: "Deployed",
+			},
+		}
+		testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, kv)
+		Eventually(func() uint32 {
+			return *clusterConfig.GetMigrationConfiguration().ParallelOutboundMigrationsPerNode
+		}).Should(BeEquivalentTo(9))
+	})
+
+	// todo: this is not working. the the wrong value is returned. Is it a bug?
```

Comment:
same like with the other todos, the config update in the components worked like this: If a bad config was supplied, return the default config, or the last valid known config.